### ISSUE TITLE
TypeScript does not flag type mismatch in Sequelize findOne method

### ISSUE THREAD

**Issue by Co-7** [sequelize/sequelize-typescript#1734]
Hello 👋, first of all, please excuse me if my issue has already been reported, but I don't know what to search for to find it, hence my issue.

I am using the following package versions :

```json
{
  "sequelize": "^6.37.3",
  "sequelize-typescript": "^2.1.6",
  "ts-node": "^10.9.2",
  "typescript": "^5.5.4",
  "typescript-eslint": "^8.0.0",
  "@eslint/js": "^9.8.0",
  "@stylistic/eslint-plugin-ts": "^2.6.1",
  "@types/eslint": "^9.6.0",
  "@types/node": "^22.1.0",
  "@types/validator": "^13.12.0",
  "@typescript-eslint/eslint-plugin": "^8.0.0",
  "@typescript-eslint/parser": "^8.0.0",
  "eslint": "^9.8.0"
}
```
The problem I'm encountering is that in my **`findOne`** method, I set a **`where`** clause on the **`username`** field and during a refactor, I changed the **`username`** to the **`id`** but forgot to change the key and only changed the value like this :
```js
const user: User = await User.findOne({
  where: {
    username: claims.id,
  },
});
```
As you can see in the screenshots, my IDE correctly indicates that **`username`** expects a string and **`claims.id`** is a number. However, I don't understand why it doesn't flag this as an issue like it usually does, for example, with **`upsert`** methods.

- Type username : 

![Image](https://github.com/user-attachments/assets/049a26cf-7adf-4316-aa86-c7de8416b5da)

- Type claims.id : 

![Image](https://github.com/user-attachments/assets/f375fe4e-362f-4c63-ac0a-4bbb865e0b9e)

Once again, I'm not sure if this is the right place to report this, but since it concerns a Sequelize method and TypeScript, I thought you might be able to help. Thank you 🙏

**Comment by Co-7** [comment]
I detected another similar problem on the findOne where typescript doesn't seem to know that the return of findOne can be null for this code : 

```js
const user = await User.findOne({
  where: {
    username: claims.id,
  },
});
```

![Image](https://github.com/user-attachments/assets/44056450-e0b4-4fcb-bd6b-60330287379d)

Despite the fact that it is well written in the vscode help: 

![Image](https://github.com/user-attachments/assets/f40c487e-0c6e-4f73-9b05-635eecaa16d7)
